src/Adapter/Order/CommandHandler/UpdateProductInOrderHandler.php

```diff
@@ -146,35 +120,15 @@ public function handle(UpdateProductInOrderCommand $command)
         // Save order detail
         $res &= $orderDetail->update();
 
-        // Update weight SUM
-        $order_carrier = new OrderCarrier((int) $order->getIdOrderCarrier());
-        if (Validate::isLoadedObject($order_carrier)) {
-            $order_carrier->weight = (float) $order->getTotalWeight();
-            $res &= $order_carrier->update();
-            if ($res) {
-                $order->weight = sprintf('%.3f ' . Configuration::get('PS_WEIGHT_UNIT'), $order_carrier->weight);
-            }
-        }
-
-        // Save order invoice
-        if (isset($orderInvoice)) {
-            $res &= $orderInvoice->update();
-        }
-
-        // Update product available quantity
-        StockAvailable::updateQuantity(
-            $orderDetail->product_id,
-            $orderDetail->product_attribute_id,
-            $old_quantity - $orderDetail->product_quantity,
-            $order->id_shop
+        // Update quantity and amounts
+        $order = $this->orderProductUpdater->update(
+            $order,
+            $orderDetail,
+            $old_quantity,
```

Comment:
My bad `$old_quantity` is till used to save the former quantity of OrderDetail
But it's only because it is updated right before this call, if the OrderDetail update is made in this function (which makes sense), then there is no more need to keep the old quantity or to pass it as a parameter to this function

Do you thin the `orderProductUpdater` should also be responsible for:
- updating the prices?
- updating the customization quantity?

If the prices is too much and would be better staying here, then maybe the `update` method should be called `updateQuantity` instead to have a clear view of its responsibility

Reply:
I just read this message. It's done. The update of product_quantity and taxes is now done in this service